commands/audit/sca/python/python.go

```diff
@@ -162,26 +228,49 @@ func installPipDeps(auditPython *AuditPython) (restoreEnv func() error, err erro
 
 	remoteUrl := ""
 	if auditPython.RemotePypiRepo != "" {
-		remoteUrl, err = utils.GetPypiRepoUrl(auditPython.Server, auditPython.RemotePypiRepo)
+		remoteUrl, err = utils.GetPypiRepoUrl(auditPython.Server, auditPython.RemotePypiRepo, auditPython.IsCurationCmd)
+		if err != nil {
+			return
+		}
+	}
+
+	var curationCachePip string
+	var reportFileName string
+	if auditPython.IsCurationCmd {
+		curationCachePip, err = xrayutils2.GetCurationPipCacheFolder()
 		if err != nil {
 			return
 		}
+		reportFileName = "report.json"
 	}
-	pipInstallArgs := getPipInstallArgs(auditPython.PipRequirementsFile, remoteUrl)
+
+	pipInstallArgs := getPipInstallArgs(auditPython.PipRequirementsFile, remoteUrl, curationCachePip, reportFileName)
+	var reqErr error
 	err = executeCommand("python", pipInstallArgs...)
 	if err != nil && auditPython.PipRequirementsFile == "" {
-		pipInstallArgs = getPipInstallArgs("requirements.txt", remoteUrl)
-		reqErr := executeCommand("python", pipInstallArgs...)
+		pipInstallArgs = getPipInstallArgs("requirements.txt", remoteUrl, curationCachePip, reportFileName)
+		reqErr = executeCommand("python", pipInstallArgs...)
 		if reqErr != nil {
 			// Return Pip install error and log the requirements fallback error.
 			log.Debug(reqErr.Error())
 		} else {
 			err = nil
 		}
 	}
+	err = errors.Join(err, curationPassThroughError(auditPython, errors.Join(err, reqErr)))
 	return
 }
 
+// If its curation command, we want to inform user that it can be resulted of pass-through disabled on curated repos.
+func curationPassThroughError(auditPython *AuditPython, errFromPip error) (err error) {
+	if !auditPython.IsCurationCmd {
+		return
+	}
+	if errFromPip != nil && strings.Contains(strings.ToLower(errFromPip.Error()), "http error 403") {
+		err = errors.New("Failed to get dependencies tree for python project, Please verify pass-through enabled on the curated repos")
+	}
+	return
+}
```

Comment:
you have similar method at `mvn.go` (`suspectCurationBlockedError`) merge those to a single method that could be use for multiple package managers and use it as util